esy-lib/posix_getumask_patch.c

```diff
@@ -0,0 +1,45 @@
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/alloc.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#if __APPLE__ || __linux__
+#else 
+#include <io.h>
+#endif
+
+
+CAMLprim value
+esy_getumask(value unit) {
+    CAMLparam1( unit );
+    CAMLlocal1( result );
+#if __APPLE__ || __linux__
+    // Reference: https://man7.org/linux/man-pages/man3/getumask.3.html
+    mode_t mask = umask( 0 );
+    umask( mask );
+    result = caml_alloc( 1, 0 );
+    Store_field( result, 0, Val_int((int) mask) ); // Ok(mask)
+#else
+    // Reference: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/umask-s?view=msvc-160#example
+    int oldmask, err, dontcare;
+
+    err = _umask_s( _S_IWRITE, &oldmask );
+    if (err) {
+        result = caml_alloc( 1, 1 );
+        char *msg = "Error setting the umask.\n";
```

Comment:
Nice work, @melwyn95 We could use `strerror` here - https://man7.org/linux/man-pages/man3/strerror.3.html

Reply:
Done